shopinvader_api_address/schemas.py

```diff
@@ -53,23 +53,19 @@ class AddressUpdate(StrictExtendableBaseModel):
     country_id: int | None = None
 
     def to_res_partner_vals(self) -> dict:
-        vals = {
-            "name": self.name,
-            "street": self.street,
-            "street2": self.street2,
-            "zip": self.zip,
-            "city": self.city,
-            "phone": self.phone,
-            "email": self.email,
-            "state_id": self.state_id,
-            "country_id": self.country_id,
-        }
+        return self.model_dump(exclude_unset=True)
```

Comment:
@lmignon : I would like your feedback here : https://github.com/shopinvader/odoo-shopinvader/issues/1427

Maybe I should move this logic in the "service" ?

Reply:
@sebastienbeau Even if in this case we've a strict mapping between declared fields on the schema and fields on the res.partner model, we must avoid to return the result of the call to `model_dump` as parameters for a call to the `write' method on the `res.partner` model. As discussed, this way you introduce a deep coupling between your schema and your model and this code will break if the schema is extended to add a field not defined on the odoo model.  Nevertheless the call to `model_dump`with the `exclude_unset=True` parameter could be good Idea to update only values set on the schema. But it's to use on purpose since you also lost default values defined on the schema (AFAIK).

```python

fields = ["name", "street", "street2", "zip", "city", "phone", "email", "state_id", "country_id"]
values = self.model_dump(exclude_unset=True)
return {f: values[f] for f in fields if f in values }
```

Reply:
This is purely mapping code, converting the result of the business logic (even if trivial in this case) to the data structure of the API. So this is good to have it in the API layer.

IMHO, mapping code is better explicit. This is may take a few seconds more to write, but it is much much easier to read afterwards. And we collectively spend much more time reading code that writing it :)

Also it is safer, because if we map fields with same names automatically, we risk accidentally exposing data when we add fields to the API data structure.